batch.go

```diff
@@ -0,0 +1,197 @@
+package loki
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	fake "github.com/brianvoe/gofakeit/v6"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/grafana/loki/pkg/logproto"
+	json "github.com/json-iterator/go"
+	"github.com/mingrammer/flog/flog"
+	"github.com/prometheus/common/model"
+	"go.k6.io/k6/lib"
+)
+
+type FakeFunc func() string
+
+type LabelPool struct {
+	Format    model.LabelValues
+	Namespace model.LabelValues
+	App       model.LabelValues
+	Pod       model.LabelValues
+	Instance  model.LabelValue
+}
+
+type Batch struct {
+	Streams   map[string]*logproto.Stream
+	Bytes     int
+	CreatedAt time.Time
+}
+
+type Entry struct {
+	logproto.Entry
+	TenantID string
+	Labels   model.LabelSet
+}
+
+func newBatch(entries ...Entry) *Batch {
+	b := &Batch{
+		Streams:   map[string]*logproto.Stream{},
+		Bytes:     0,
+		CreatedAt: time.Now(),
+	}
+
+	// Add entries to the batch
+	for _, entry := range entries {
+		b.add(entry)
+	}
+
+	return b
+}
+
+// add an entry to the batch
+func (b *Batch) add(entry Entry) {
+	b.Bytes += len(entry.Line)
+
+	// Append the entry to an already existing stream (if any)
+	labels := entry.Labels.String()
+	if stream, ok := b.Streams[labels]; ok {
+		stream.Entries = append(stream.Entries, entry.Entry)
+		return
+	}
+
+	// Add the entry as a new stream
+	b.Streams[labels] = &logproto.Stream{
+		Labels:  labels,
+		Entries: []logproto.Entry{entry.Entry},
+	}
+}
+
+// sizeBytes returns the current batch size in bytes
+func (b *Batch) sizeBytes() int {
+	return b.Bytes
+}
+
+// sizeBytesAfter returns the size of the batch after the input entry
+// will be added to the batch itself
+func (b *Batch) sizeBytesAfter(entry Entry) int {
+	return b.Bytes + len(entry.Line)
+}
+
+// age of the batch since its creation
+func (b *Batch) age() time.Duration {
+	return time.Since(b.CreatedAt)
+}
```

Comment:
These function are unused at the moment.

Reply:
I don't think we'll need it.